### PR TITLE
DomainAwareDataset str repr edge case handling

### PR DESCRIPTION
**Pull request by YanisLalou**
Issue #12 
- Handling the str output of a DomainAwareDataset if the number of domin_names_ is too large
- Print the number of datapoints in the dataset

### CODE REVIEW

**Comment by YanisLalou** [comment]
The goal is to have a similar string representation as torchvision for datasets:
`Dataset CIFAR10
    Number of datapoints: 50000
    Root location: datasets/CIFAR10
    Split: Train`

However rn in the `DomainAwareDataset` object we don't have the Dataset name + Root location

File: skada/datasets/_base.py [hunk 1]
```diff
@@ -314,10 +314,31 @@ def pack_lodo(self, return_X_y: bool = True) -> PackedDatasetType:
...
 
     def __repr__(self) -> str:
-        return self.__str__()
+        head = self.__str__()
+        body = [f"Number of domains: {len(self.domains_)}"]
+        body.append(f"Total size: {sum(len(tup[0]) for tup in self.domains_)}")
+        output = "\n".join([head] + body)
+        return output
+    
+    def get_domain_representation(self, max_domains=5, max_length=50):
```
Comment (kachayev) [inline skada/datasets/_base.py:326]: I suggest renaming the function to `_get_domain_representation` to emphasize its role as an internal function. This change will help clarify that it should not be considered part of the class's public API.

**Review by rflamary — COMMENTED** [review]
This seems OK but you need to do some testing (add at leats one test) with different types of datasets and check that nothing breaks

**Comment by kachayev** [comment]
@YanisLalou Flake8 is not happy here as well.

**Pull request merged**